backend/service/chaos/experimentation/experimentstore/experiment_store.go

```diff
@@ -58,61 +50,66 @@ func New(cfg *any.Any, _ *zap.Logger, _ tally.Scope) (service.Service, error) {
 }
 
 func (fs *experimentStore) CreateExperiments(ctx context.Context, experiments []*experimentation.Experiment) error {
-	psql := sq.StatementBuilder.PlaceholderFormat(sq.Dollar)
-	builder := psql.Insert(experimentsTableName).Columns(experimentColumns...)
+	// This API call will eventually be broken into 2 separate calls:
+	// 1) creating the config
+	// 2) starting a new experiment with the config
+
+	// All experiments are created in a single transaction
+	tx, err := fs.db.Begin()
+	if err != nil {
+		return err
+	}
 
 	for _, experiment := range experiments {
-		marshaler := jsonpb.Marshaler{}
-		buf := &bytes.Buffer{}
-		err := marshaler.Marshal(buf, experiment.GetConfig())
+		// Step 1) create the config
+		configID := id.NewID()
+
+		configJson, err := marshalConfig(experiment)
 		if err != nil {
 			return err
 		}
-		s := buf.String()
-		builder = builder.Values(
-			id.NewID(),
-			s,
-		)
-	}
 
-	sql, args, err := builder.ToSql()
-	if err != nil {
-		return err
+		configSql := `INSERT INTO experiment_config (id, details) VALUES ($1, $2)`
+		_, err = fs.db.ExecContext(ctx, configSql, configID, configJson)
+		if err != nil {
+			return err
+		}
+
+		// Step 2) start a new experiment with the config
+		runSql := `
+			INSERT INTO experiment_run (
+                id, 
+		        experiment_config_id,
+		        start_time,
+			    end_time,                        
+		        creation_time)
+            VALUES ($1, $2, NOW(), NULL, NOW())`
+
+		_, err = fs.db.ExecContext(ctx, runSql, id.NewID(), configID)
```

Comment:
for `start_time` and end_time (`scheduled_end_time`) - can we add an option to pass it as part of the `CreateExperiments` method call - both of them would be optional so we could have the existing fallbacks in place (setting `start_time` to current time if it's not provided from the outside)? I know that they are not used by `experimentation` as of now but adding them now would allow us to avoid a follow up change in Clutch to unlock other projects.

Reply:
Yeah, we should do that in the followup API change.

Reply:
https://jira.lyft.net/browse/RES-1369